"-Werror=conversion" error after merging PR #3718 in Linux

Building yuzu on master and mainline in Linux, I get the following errors:
```
In file included from yuzu/src/./video_core/engines/kepler_compute.h:10,
                 from yuzu/src/video_core/renderer_vulkan/vk_rasterizer.cpp:20:
yuzu/src/./common/bit_field.h: In instantiation of ‘constexpr void BitField<Position, Bits, T, EndianTag>::Assign(const T&) [with long unsigned int Position = 0; long unsigned int Bits = 1; T = short unsigned int; EndianTag = KeepTag]’:
yuzu/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:131:51:   required from here
yuzu/src/./common/bit_field.h:183:63: error: conversion from ‘int’ to ‘BitField<0, 1, short unsigned int>::StorageTypeWithEndian’ {aka ‘short unsigned int’} may change value [-Werror=conversion]
  183 |         storage = (static_cast<StorageType>(storage) & ~mask) | FormatValue(value);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
```
```
In file included from yuzu/src/./video_core/renderer_vulkan/vk_rasterizer.h:20,
                 from yuzu/src/video_core/renderer_vulkan/renderer_vulkan.cpp:31:
yuzu/src/./video_core/renderer_vulkan/fixed_pipeline_state.h: In member function ‘void Vulkan::FixedPipelineState::VertexInput::SetBinding(std::size_t, bool, u32, u32)’:
yuzu/src/./video_core/renderer_vulkan/fixed_pipeline_state.h:132:35: error: conversion from ‘u32’ {aka ‘unsigned int’} to ‘short unsigned int’ may change value [-Werror=conversion]
  132 |             binding.stride.Assign(stride);
      |                                   ^~~~~~
```

The second was obtained by modifying the first error to `storage = (u_int16_t)(static_cast<StorageType>(storage) & ~mask) | (u_int16_t)FormatValue(value);`. If I modify the second to read `binding.stride.Assign((u_int16_t)stride);`, the build will complete, but games softlock immediately, so there is something I don't understand well enough to fix the error correctly.

My system specs are:
Manjaro Linux 20.0
Kernel 5.6.5-10-tkg-pds
Ryzen 7 2700X
gcc (Arch Linux 9.3.0-1) 9.3.0

The error is NOT detected by the Linux:MingW Docker build environment (`-DENABLE_VULKAN=OFF` is a CMake argument in `.ci/scripts/windows/docker.sh`). I do not have the Linux Docker environment setup.

--- a/src/common/bit_field.h
+++ b/src/common/bit_field.h
@@ -180,7 +180,7 @@ public:
     }
 
     constexpr void Assign(const T& value) {
-        storage = (static_cast<StorageType>(storage) & ~mask) | FormatValue(value);
+        storage = static_cast<StorageType>((storage & ~mask) | FormatValue(value));
     }
 
     constexpr T Value() const {

--- a/src/video_core/engines/shader_bytecode.h
+++ b/src/video_core/engines/shader_bytecode.h
@@ -655,6 +655,7 @@ union Instruction {
     }
 
     constexpr Instruction(u64 value) : value{value} {}
+    constexpr Instruction(const Instruction& instr) : value(instr.value) {}
 
     BitField<0, 8, Register> gpr0;
     BitField<8, 8, Register> gpr8;

--- a/src/video_core/renderer_vulkan/fixed_pipeline_state.h
+++ b/src/video_core/renderer_vulkan/fixed_pipeline_state.h
@@ -129,7 +129,7 @@ struct alignas(32) FixedPipelineState {
             auto& binding = bindings[index];
             binding.raw = 0;
             binding.enabled.Assign(enabled ? 1 : 0);
-            binding.stride.Assign(stride);
+            binding.stride.Assign(static_cast<u16>(stride));
             binding_divisors[index] = divisor;
         }
 

